features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/internal/data/TraceWriterTest.kt

```diff
@@ -333,4 +328,12 @@ internal class TraceWriterTest {
     }
 
     // endregion
+
+    private fun generateSpanListWithPriorities(forge: Forge, priorities: Set<Int>): MutableList<DDSpan> {
+        return forge.aList {
+            SpanForgeryFactory { frg, span ->
+                span.samplingPriority = frg.anElementFrom(priorities)
+            }.getForgery(forge)
+        }.toMutableList()
+    }
```

Comment:
it can be done simpler, without the need to do changes in `SpanForgeryFactory` (plus it can be an extension method):

```suggestion
    private fun Forge.spansWithPriorities(forge: Forge, priorities: Set<Int>): MutableList<DDSpan> {
        return aList {
            getForgery<DDSpan>().apply { span.samplingPriority = anElementFrom(priorities) }
        }.toMutableList()
    }
```

but I'm okay with keeping the proposed change as well.

Reply:
> it can be done simpler

I did it the way you describe at first. But this actually is a different solution that didn't quite work. 

1. Suppose `SpanForgeryFactory` sets `samplingPriorityHere` to `PrioritySampling.USER_DROP` [here](https://github.com/DataDog/dd-sdk-android/blob/f807ef6c3e71a3580e6a80f2bee0f28904d02d9e/features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/utils/forge/SpanForgeryFactory.kt#L39).
2. Then we change it the way you describe to `PrioritySampling.UNSET`. `PrioritySampling.UNSET` satisfies the condition `!in DROP_SAMPLING_PRIORITIES`, it was used in the test before, so we probably should not forget about it.
3. [Here](https://github.com/DataDog/dd-sdk-android/blob/f807ef6c3e71a3580e6a80f2bee0f28904d02d9e/features/dd-sdk-android-trace/src/main/java/com/datadog/opentracing/DDSpanContext.java#L222) `PrioritySampling.UNSET` is ignored.
4. We end up with a failed test because we have `PrioritySampling.USER_DROP` as priority.

Reply:
okay, makes sense then for `PrioritySampling.UNSET` case